### ISSUE TITLE
youtube api v2.0 sunset. Need to update to use v3.0 api.

### ISSUE THREAD

**Issue by pdenney** [0xced/XCDYouTubeKit#142]
Currently, the preview window for the images gets an error pointing to this page:
https://support.google.com/youtube/answer/6098135?p=yt_devicesupport&hl=en&rd=1

The api being used needs to be updated to use the v3.0 api.

**Comment by waxx91** [comment]
i am having the same issue. what is the solution?

**Comment by 0xced** [comment]
XCDYouTubeKit is not using the official YouTube API and is not affected by this change.

**Comment by pdenney** [comment]
@0xced that is not true, you use it when you call https://www.youtube.com/get_video_info? in XCDYouTubeVideoOperation. Currently, it is causing an error message for all thumbnails that are parsed from the response. In the future, Google is going to sunset this call and even the http streams you parse from there will be gone. 

I looked for similar functionality in v3 of the api to fix it and submit a pull request but there isn't a 1-to-1 correlation. While we can fetch the thumbnails from the v3 api, we can't fetch the streams.

**Comment by 0xced** [comment]
> Currently, it is causing an error message for all thumbnails that are parsed from the response.

I don’t understand this at all. First, I don’t get an error message. Second, all the thumbnails that I tested are working fine. For example, for video with identifier `9bZkp7q19f0` I get these thumbnail URLs:
Small  thumbnail: https://i.ytimg.com/vi/9bZkp7q19f0/default.jpg
Medium thumbnail: https://i.ytimg.com/vi/9bZkp7q19f0/sddefault.jpg
Large  thumbnail: https://i.ytimg.com/vi/9bZkp7q19f0/maxresdefault.jpg

I have tested with several other videos and all thumbnails are fine. I think those error messages you see are not coming from XCDYouTubeKit, someone already did this mistake in #136. If those messages are coming from XCDYouTubeKit, I’d appreciate a sample app demonstrating the issue, as suggested in the [contributing guidelines](https://github.com/0xced/XCDYouTubeKit/blob/master/CONTRIBUTING.md).

Also, note that the `get_video_info` endpoint is NOT the YouTube API v2. Fetching the streams was never part of any YouTube API, neither v2 nor v3.
